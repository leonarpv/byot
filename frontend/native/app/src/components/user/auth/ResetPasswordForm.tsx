import * as React from 'react';
import {Text, View} from 'react-native-ui-lib';
import {Formik} from 'formik';
import {useTranslation} from 'react-i18next';
import {IResetPassword} from '@byot-frontend/common/src/types/interfaces/IResetPassword';
import {ResetPassword} from '@byot-frontend/common/src/types/dto/ResetPassword';
import {resetPasswordSchema} from '@byot-frontend/common/src/types/schemas/validation/ResetPasswordSchema';
import {TextField} from '../../elements/lib/TextField';
import {Button} from '../../elements/lib/Button';

export const testIDs = {
  newPassword: 'resetPassword-newPassword',
  passwordRepeat: 'resetPassword-passwordRepeat',
  submit: 'resetPassword-submit',
};

interface Props {
  loading: boolean;
  onSubmit: (values: IResetPassword) => void;
  token: string;
}

export const ResetPasswordForm: React.FC<Props> = (props: Props) => {
  const {t} = useTranslation();
  const [password, setPassword] = React.useState('');
  const initialValues: IResetPassword = new ResetPassword({token: props.token});
  return (
    <>
      <Text heading>{t('Reset password')}</Text>
      <Formik
        initialValues={initialValues}
        validateOnBlur
        validateOnChange
        validationSchema={resetPasswordSchema(t, password)}
        onSubmit={values => props.onSubmit(values)}>
        {({values, handleChange, handleSubmit, handleBlur, errors, touched}) => (
          <>
            <View marginB-10>
              <TextField
                placeholder={t('Enter new password')}
                value={values.newPassword}
                secureTextEntry
                onBlur={handleBlur('newPassword')}
                onChangeText={text => {
                  setPassword(text);
                  handleChange('newPassword')(text);
                }}
                infoText
                error={(touched.newPassword && errors.newPassword) as string}
                testID={testIDs.newPassword}
                accessibilityLabel={testIDs.newPassword}
                autoCapitalize="none"
              />
            </View>
            <View marginB-10>
              <TextField
                placeholder={t('Confirm password')}
                value={values.passwordRepeat}
                secureTextEntry
                onBlur={handleBlur('passwordRepeat')}
                onChangeText={handleChange('passwordRepeat')}
                infoText
                error={(touched.passwordRepeat && errors.passwordRepeat) as string}
                testID={testIDs.passwordRepeat}
                accessibilityLabel={testIDs.passwordRepeat}
                autoCapitalize="none"
              />
            </View>
            <View marginB-10>
              <Button
                label={t('Reset password')}
                testID={testIDs.submit}
                onPress={handleSubmit}
                loading={props.loading}
              />
            </View>
          </>
        )}
      </Formik>
    </>
  );
};
