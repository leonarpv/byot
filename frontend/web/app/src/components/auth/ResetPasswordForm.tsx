import * as React from 'react';
import {Grid} from '@material-ui/core';
import {TFunction} from 'i18next';
import {Form, Formik} from 'formik';
import * as Yup from 'yup';
import {PlainLayoutNarrow} from '@byot-frontend/web-common/src/components/plain-layout/PlainLayoutNarrow';
import {PlainLayoutTitle} from '@byot-frontend/web-common/src/components/plain-layout/PlainLayoutTitle';
import {useTranslation} from 'react-i18next';
import {Input} from '@byot-frontend/web-common/src/components/elementary/form/Input';
import {Button} from '@byot-frontend/web-common/src/components/elementary/form/Button';
import {IResetPassword} from '@byot-frontend/common/src/types/interfaces/IResetPassword';
import {ResetPassword} from '@byot-frontend/common/src/types/dto/ResetPassword';

const resetPassSchema = (t: TFunction, password: string) =>
  Yup.object().shape({
    newPassword: Yup.string()
      .required(t('Enter password'))
      .min(8, t('Password must be longer than 8 characters'))
      .matches(/[^A-Za-z0-9]+/, t('Password must contain at least one special character')),
    passwordRepeat: Yup.string()
      .required(t('Confirm password'))
      .oneOf([password], t('Passwords are not the same')),
  });

interface Props {
  loading: boolean;
  onSubmit: (values: IResetPassword) => void;
  token: string;
}

export const ResetPasswordForm: React.FC<Props> = (props: Props) => {
  const {t} = useTranslation();
  const [password, setPassword] = React.useState('');
  const initialValues: IResetPassword = new ResetPassword({token: props.token});
  return (
    <PlainLayoutNarrow>
      <PlainLayoutTitle>{t('Reset password')}</PlainLayoutTitle>
      <Formik
        initialValues={initialValues}
        validateOnBlur
        validateOnChange
        validationSchema={resetPassSchema(t, password)}
        onSubmit={values => props.onSubmit(values)}>
        {({values, handleChange, handleBlur, errors, touched}) => (
          <Form data-testid="common-auth-resetPassword-form-form">
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <Input
                  data-testid="common-auth-resetPassword-form-newPassword"
                  label={t('Enter new password')}
                  error={!!(touched.newPassword && errors.newPassword)}
                  helperText={touched.newPassword && errors.newPassword}
                  onChange={e => {
                    setPassword(e.target.value);
                    handleChange('newPassword')(e);
                  }}
                  onBlur={handleBlur('newPassword')}
                  value={values.newPassword}
                  type="password"
                />
              </Grid>
              <Grid item xs={12}>
                <Input
                  data-testid="common-auth-resetPassword-form-passwordRepeat"
                  label={t('Confirm password')}
                  error={!!(touched.passwordRepeat && errors.passwordRepeat)}
                  helperText={touched.passwordRepeat && errors.passwordRepeat}
                  onChange={handleChange('passwordRepeat')}
                  onBlur={handleBlur('passwordRepeat')}
                  value={values.passwordRepeat}
                  type="password"
                />
              </Grid>
              <Grid item xs={12}>
                <Button type="submit" fullWidth loading={props.loading}>
                  {t('Reset password')}
                </Button>
              </Grid>
            </Grid>
          </Form>
        )}
      </Formik>
    </PlainLayoutNarrow>
  );
};
