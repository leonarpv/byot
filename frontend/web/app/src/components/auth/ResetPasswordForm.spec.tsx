import * as React from 'react';
import {fireEvent, render, act, waitFor} from '@testing-library/react';
import {ResetPasswordForm} from './ResetPasswordForm';

describe('<ResetPasswordForm/>', () => {
  it('should render', () => {
    const {container} = render(<ResetPasswordForm onSubmit={jest.fn()} loading={false} token="token" />);
    expect(container).toMatchSnapshot();
  });
  it('should not accept simple password', async () => {
    const onSubmit = jest.fn();
    const {container, getByTestId} = render(
      <ResetPasswordForm onSubmit={onSubmit} loading={false} token="token" />
    );
    const newPassword = getByTestId('common-auth-resetPassword-form-newPassword').querySelector('input');
    const passwordRepeat = getByTestId('common-auth-resetPassword-form-passwordRepeat').querySelector(
      'input'
    );
    const button = getByTestId('common-auth-resetPassword-form-form').querySelector('button[type="submit"]');
    expect(button).not.toBeNull();
    expect(newPassword).not.toBeNull();
    expect(passwordRepeat).not.toBeNull();
    fireEvent.change(newPassword!, {target: {value: 'a'}});
    fireEvent.change(passwordRepeat!, {target: {value: 'a'}});
    fireEvent.click(button!);
    await waitFor(() => expect(onSubmit).not.toBeCalled());
    expect(container).toMatchSnapshot();
  });
  it('should not accept different passwords', async () => {
    const onSubmit = jest.fn();
    const {container, getByTestId} = render(
      <ResetPasswordForm onSubmit={onSubmit} loading={false} token="token" />
    );
    const newPassword = getByTestId('common-auth-resetPassword-form-newPassword').querySelector('input');
    const passwordRepeat = getByTestId('common-auth-resetPassword-form-passwordRepeat').querySelector(
      'input'
    );
    const button = getByTestId('common-auth-resetPassword-form-form').querySelector('button[type="submit"]');
    expect(newPassword).not.toBeNull();
    expect(passwordRepeat).not.toBeNull();
    expect(button).not.toBeNull();
    fireEvent.change(newPassword!, {target: {value: '123-Abc$'}});
    fireEvent.change(passwordRepeat!, {target: {value: '123-AbcS'}});
    fireEvent.click(button!);
    await waitFor(() => expect(onSubmit).not.toBeCalled());
    expect(container).toMatchSnapshot();
  });
  it('should accept same difficult passwords', async () => {
    const onSubmit = jest.fn();
    const {container, getByTestId} = render(
      <ResetPasswordForm onSubmit={onSubmit} loading={false} token="token" />
    );
    const newPassword = getByTestId('common-auth-resetPassword-form-newPassword').querySelector('input');
    const passwordRepeat = getByTestId('common-auth-resetPassword-form-passwordRepeat').querySelector(
      'input'
    );
    const button = getByTestId('common-auth-resetPassword-form-form').querySelector('button[type="submit"]');
    expect(button).not.toBeNull();
    expect(newPassword).not.toBeNull();
    expect(passwordRepeat).not.toBeNull();
    fireEvent.change(newPassword!, {target: {value: '123-Abc$'}});
    fireEvent.change(passwordRepeat!, {target: {value: '123-Abc$'}});
    fireEvent.click(button!);
    await waitFor(() =>
      expect(onSubmit).toBeCalledWith({
        newPassword: '123-Abc$',
        passwordRepeat: '123-Abc$',
        token: 'token',
      })
    );
    expect(container).toMatchSnapshot();
  });
});
