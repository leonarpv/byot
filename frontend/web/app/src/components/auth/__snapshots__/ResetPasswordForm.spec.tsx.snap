// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResetPasswordForm/> should accept same difficult passwords 1`] = `
<div>
  <div
    class="makeStyles-root-34"
  >
    <div
      class="MuiBox-root MuiBox-root-35"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Reset password
      </h3>
    </div>
    <form
      action="#"
      data-testid="common-auth-resetPassword-form-form"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-newPassword"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-37 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
              data-shrink="true"
            >
              Enter new password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-36 makeStyles-filledRoot-38 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value="123-Abc$"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained"
                >
                  <span>
                    ​
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-passwordRepeat"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-37 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
              data-shrink="true"
            >
              Confirm password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-36 makeStyles-filledRoot-39 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value="123-Abc$"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container"
            style="min-height: 0px; height: 0px; transition-duration: 300ms;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained"
                >
                  <span>
                    ​
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-40 makeStyles-root-43 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-inner-41 makeStyles-inner-44"
                data-testid="common-elementary-button-children"
                style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
              >
                Reset password
              </span>
              <div
                class="makeStyles-loader-42"
                data-testid="common-elementary-button-loader"
                style="opacity: 0; visibility: hidden;"
              >
                <div
                  class="css-0"
                >
                  <div
                    class="css-1ab95gp"
                  />
                  <div
                    class="css-tm4xur"
                  />
                  <div
                    class="css-1epl0i6"
                  />
                  <div
                    class="css-1fj2adh"
                  />
                  <div
                    class="css-1vivfwt"
                  />
                </div>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<ResetPasswordForm/> should not accept different passwords 1`] = `
<div>
  <div
    class="makeStyles-root-23"
  >
    <div
      class="MuiBox-root MuiBox-root-24"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Reset password
      </h3>
    </div>
    <form
      action="#"
      data-testid="common-auth-resetPassword-form-form"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-newPassword"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-26 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
              data-shrink="true"
            >
              Enter new password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-25 makeStyles-filledRoot-27 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value="123-Abc$"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained"
                >
                  <span>
                    ​
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-passwordRepeat"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-26 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-error Mui-error MuiFormLabel-filled"
              data-shrink="true"
            >
              Confirm password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-25 makeStyles-filledRoot-28 Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="true"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value="123-AbcS"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container"
            style="min-height: 0px; height: 0px; transition-duration: 300ms;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
                >
                  Passwords are not the same
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-29 makeStyles-root-32 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-inner-30 makeStyles-inner-33"
                data-testid="common-elementary-button-children"
                style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
              >
                Reset password
              </span>
              <div
                class="makeStyles-loader-31"
                data-testid="common-elementary-button-loader"
                style="opacity: 0; visibility: hidden;"
              >
                <div
                  class="css-0"
                >
                  <div
                    class="css-1ab95gp"
                  />
                  <div
                    class="css-tm4xur"
                  />
                  <div
                    class="css-1epl0i6"
                  />
                  <div
                    class="css-1fj2adh"
                  />
                  <div
                    class="css-1vivfwt"
                  />
                </div>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<ResetPasswordForm/> should not accept simple password 1`] = `
<div>
  <div
    class="makeStyles-root-12"
  >
    <div
      class="MuiBox-root MuiBox-root-13"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Reset password
      </h3>
    </div>
    <form
      action="#"
      data-testid="common-auth-resetPassword-form-form"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-newPassword"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-15 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled Mui-error Mui-error MuiFormLabel-filled"
              data-shrink="true"
            >
              Enter new password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-14 makeStyles-filledRoot-16 Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="true"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value="a"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container"
            style="min-height: 0px; height: 0px; transition-duration: 300ms;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
                >
                  Password must be longer than 8 characters
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-passwordRepeat"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-15 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
              data-shrink="true"
            >
              Confirm password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-14 makeStyles-filledRoot-17 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value="a"
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container"
            style="min-height: 0px; height: 0px; transition-duration: 300ms;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained"
                >
                  <span>
                    ​
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-18 makeStyles-root-21 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-inner-19 makeStyles-inner-22"
                data-testid="common-elementary-button-children"
                style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
              >
                Reset password
              </span>
              <div
                class="makeStyles-loader-20"
                data-testid="common-elementary-button-loader"
                style="opacity: 0; visibility: hidden;"
              >
                <div
                  class="css-0"
                >
                  <div
                    class="css-1ab95gp"
                  />
                  <div
                    class="css-tm4xur"
                  />
                  <div
                    class="css-1epl0i6"
                  />
                  <div
                    class="css-1fj2adh"
                  />
                  <div
                    class="css-1vivfwt"
                  />
                </div>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<ResetPasswordForm/> should render 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-2"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3"
      >
        Reset password
      </h3>
    </div>
    <form
      action="#"
      data-testid="common-auth-resetPassword-form-form"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-newPassword"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Enter new password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-3 makeStyles-filledRoot-5 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained"
                >
                  <span>
                    ​
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
            data-testid="common-auth-resetPassword-form-passwordRepeat"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root makeStyles-label-4 MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
              data-shrink="false"
            >
              Confirm password
            </label>
            <div
              class="MuiInputBase-root MuiFilledInput-root makeStyles-filledRoot-3 makeStyles-filledRoot-6 MuiInputBase-fullWidth MuiInputBase-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiFilledInput-input"
                type="password"
                value=""
              />
            </div>
          </div>
          <div
            class="MuiCollapse-container MuiCollapse-hidden"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <p
                  class="MuiFormHelperText-root MuiFormHelperText-contained"
                >
                  <span>
                    ​
                  </span>
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <button
            class="MuiButtonBase-root MuiButton-root makeStyles-root-7 makeStyles-root-10 MuiButton-contained MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="makeStyles-inner-8 makeStyles-inner-11"
                data-testid="common-elementary-button-children"
                style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
              >
                Reset password
              </span>
              <div
                class="makeStyles-loader-9"
                data-testid="common-elementary-button-loader"
                style="opacity: 0; visibility: hidden;"
              >
                <div
                  class="css-0"
                >
                  <div
                    class="css-1ab95gp"
                  />
                  <div
                    class="css-tm4xur"
                  />
                  <div
                    class="css-1epl0i6"
                  />
                  <div
                    class="css-1fj2adh"
                  />
                  <div
                    class="css-1vivfwt"
                  />
                </div>
              </div>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
